presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1545,6 +1554,17 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
         return result;
     }
 
+    private HiveCompressionCodec getHiveCompressionCodec(ConnectorSession session, boolean isTemporaryTable, HiveStorageFormat storageFormat)
+    {
+        if (isTemporaryTable) {
+            return getTemporaryTableCompressionCodec(session);
+        }
+        if (storageFormat.equals(ORC) || storageFormat.equals(DWRF)) {
+            return getORCCompressionCodec(session);
+        }
+        return getCompressionCodec(session);
```

Comment:
What other storage format cannot have zstd?

Reply:
Anything other than ORC and DWRF.